src/Symfony/Component/Serializer/Mapping/AttributeMetadata.php

```diff
@@ -110,24 +114,66 @@ public function getMaxDepth(): ?int
         return $this->maxDepth;
     }
 
-    public function setSerializedName(?string $serializedName): void
+    public function setSerializedName(?string $serializedName /* , array $groups = [] */): void
+    {
+        $groups = 2 <= \func_num_args() ? func_get_arg(1) : [];
```

Comment:
```suggestion
        $groups = \func_get_args()[1] ?? [];
```
(same for `getSerializedName` and for serialized path related methods)

Or what about
```suggestion
        $groups = \func_get_args()[1] ?? ['*'];
```
?
If so, we must update the parameter default value in PHPDocs

Reply:
I followed [official recommendations](https://symfony.com/doc/current/contributing/code/bc.html#:~:text=Use%20func_num_args%20and%20func_get_arg%20to%20retrieve%20the%20argument%20in%20the%20method%3A) on retrieving undeclared arguments.
Also, looks like entire codebase follow these recommendations too. Example:
https://github.com/symfony/symfony/blob/ce8be2916117310ad3cdbb012b70231f29b57591/src/Symfony/Component/Serializer/NameConverter/CamelCaseToSnakeCaseNameConverter.php#L60-L62
Maybe we should keep current style for consistency (at least for now)?

Reply:
As for default group `['*']`, I'm ok with that👍